src/quart_schema/extension.py

```diff
@@ -442,17 +459,19 @@ def _build_openapi_schema(app: Quart, extension: QuartSchema) -> dict:
             components["schemas"].update(definitions)
             for name, type_ in schema["properties"].items():
                 param = {"name": name.replace("_", "-"), "in": "header", "schema": type_}
-                if "description" in type_:
-                    param["description"] = type_.pop("description")
 
-                operation_object["parameters"].append(param)  # type: ignore
+                for attribute in ("description", "required", "deprecated"):
```

Comment:
I'm not sure about required here, isn't every parameter that isn't `thing: Optional[type] = None` required?

Reply:
Yes, but that's besides the point here. All the attributes that come from the pydantic-generated schema are by default added to the `schema` attribute of the parameter in the generated openapi code. These three attributes have to be on the top level of the parameter definition by the openapi spec (and for swagger et al to pick up on it).